Remove all unused variables

arielsvn opened the pull request:
## Issue Number

N/A

## Purpose/Implementation Notes

Initially suggested in https://github.com/AlexsLemonade/refinebio/pull/2034#issuecomment-570360849

This uses `autoflake` to remove all unused variables in the project.

I executed the following commands 

```
$ autoflake -r --in-place --remove-unused-variables api/
$ autoflake -r --in-place --remove-unused-variables common/
$ autoflake -r --in-place --remove-unused-variables foreman 
$ autoflake -r --in-place --remove-unused-variables infrastructure 
$ autoflake -r --in-place --remove-unused-variables workers 
```

## Types of changes

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Functional tests

None

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

cgreene commented on foreman/data_refinery_foreman/surveyor/geo.py:
```diff
@@ -116,11 +115,11 @@ def set_platform_properties(
             # Example of it being a list: GSE113945
             channel1_temp = sample_metadata.get("label_protocol_ch1", "")
             if type(channel1_temp) != list:
```
This looks like a bug. I imagine we were hoping to do something with channel1_protocol (otherwise, why are we doing any of this?).

arielsvn replied:
looks like we could remove the whole `if...` since `channel1_temp` is also not used below

kurtwheeler replied:
If I had to guess I'd guess it had to with https://github.com/AlexsLemonade/refinebio/issues/87 or https://github.com/AlexsLemonade/refinebio/issues/89

cgreene commented on api/data_refinery_api/tests.py:
```diff
@@ -629,7 +628,7 @@ def test_processed_samples_only(self):
         self.assertEqual(response.json()["count"], 1)
 
-        qs = Experiment.processed_public_objects
+        Experiment.processed_public_objects
```
Bug? Looks like we would have intended to use this.

arielsvn replied:
I think we can safetly remove the whole line, since `qs` wasn't used here. And we don't need to test `Experiment.processed_public_objects`

cgreene commented on common/data_refinery_common/migrations/0045_svd_algorithm.py:
```diff
@@ -3,7 +3,7 @@
 def svd_algorithm_fix(apps, schema_editor):
     """ Fetch existing compendia and create compendium results"""
-    CompendiumResult = apps.get_model("data_refinery_common", "CompendiumResult")
+    apps.get_model("data_refinery_common", "CompendiumResult")
```
Either a bug or dead code?

arielsvn replied:
dead code, will remove the whole line

cgreene commented on foreman/data_refinery_foreman/foreman/management/commands/run_tximport.py:
```diff
@@ -77,7 +74,7 @@ def run_tximport():
                 # in that it doesn't actuallhy use original files so
                 # this is just used to point to the experiment.
-                samples = experiment.samples.all()
+                experiment.samples.all()
```
This was a dead query. Did we want it to do something?

arielsvn replied:
looks like we wanted to use it below but added the whole value instead. Will remove the whole line

cgreene commented on foreman/data_refinery_foreman/foreman/management/commands/test_retry_samples.py:
```diff
@@ -38,7 +34,7 @@ def setup_experiment() -> Dict:
     experiment_accession = "SRP095529"
     data_dir = "/home/user/data_store/"
-    experiment_dir = data_dir + experiment_accession
+    data_dir + experiment_accession
```
This isn't getting used. Also looks like a bug.

cgreene commented on foreman/data_refinery_foreman/foreman/test_main.py:
```diff
@@ -292,7 +291,7 @@ def mock_init_nomad(host, port=0, timeout=0):
         self.assertEqual(len(mock_send_job.mock_calls), 0)
 
-        jobs = DownloaderJob.objects.order_by("id")
+        DownloaderJob.objects.order_by("id")
```
Dead code here (and later in this file) all looks safe to remove.

cgreene commented on foreman/data_refinery_foreman/surveyor/test_harmony.py:
```diff
@@ -215,7 +210,7 @@ def test_sdrf_big(self):
                 continue
             # No assertions, just making sure we don't barf.
-            harmonized = harmonize(metadata)
+            harmonize(metadata)
```
This actually looks right

cgreene commented on foreman/data_refinery_foreman/surveyor/test_harmony.py:
```diff
@@ -270,8 +265,8 @@ def test_sra_lots(self):
                 metadata = SraSurveyor.gather_all_metadata(accession)
-                harmonized = harmonize([metadata])
-            except UnsupportedDataTypeError as udte:
+                harmonize([metadata])
```
On the other hand - not sure what this pattern is trying to do. Run harmonize but then we're ok with UnsupportedDataTypeError? Looks like not a great test pattern. Seems better to test one where we expect success and another where we assert that this error is raised.

arielsvn replied:
I agree, I think we should add this on a different PR and leave this one only for unused variables.

cgreene commented on foreman/data_refinery_foreman/surveyor/transcriptome_index.py:
```diff
@@ -313,7 +313,7 @@ def _generate_files(self, species: Dict) -> None:
         gtf_download_url = url_builder.build_gtf_url()
 
-        platform_accession_code = species.pop("division")
+        species.pop("division")
```
It seems like we would have wanted this? Maybe bug?

arielsvn replied:
hmm not sure why we do this, looks like `species` is no longer used in this method after this.

kurtwheeler replied:
Hmm yeah this is odd, not sure why we'd call this platform_accession_code. It's also not used after the method either. Probably safe to remove the line entirely.

arielsvn replied:
and also the call below to `self._clean_metadata(species)`? Since `species` is no longer used here.

kurtwheeler replied:
Yes. My guess is we used more things from the species dict for the original file metadata before but now that's handled by the URL builder classes.

arielsvn replied:
ok removed `self._clean_metadata`. Will let the tests pass before merging.

cgreene commented on workers/data_refinery_workers/processors/illumina.py:
```diff
@@ -319,7 +317,7 @@ def _run_illumina(job_context: Dict) -> Dict:
     and the column identifiers which the R script needs for processing.
     """
-    input_file_path = job_context["input_file_path"]
+    job_context["input_file_path"]
```
It seems like we have a bug here or dead code.

arielsvn replied:
dead code, will remove the whole line

cgreene commented on workers/data_refinery_workers/processors/management/commands/run_tximport.py:
```diff
@@ -67,6 +67,6 @@ def handle(self, *args, **options):
         pjofa.save()
 
-        final_context = tximport.tximport(job.id)
+        tximport.tximport(job.id)
```
It seems like we'd probably want that final_context...

arielsvn replied:
for what? doesn't look like is needed

kurtwheeler replied:
I think most of instances of `final_context` were relics from Rich functional testing his code. Occasionally they'd be used in unit tests as well, but not always.

cgreene commented on workers/data_refinery_workers/processors/test_salmon.py:
```diff
@@ -236,7 +234,7 @@ def test_no_salmon_on_geo(self):
         sample_object.save()
 
-        job_context = salmon.salmon(job.pk)
+        salmon.salmon(job.pk)
```
Job context goes unused through here. Probably ok.

cgreene commented on workers/data_refinery_workers/processors/test_smasher.py:
```diff
@@ -560,7 +559,7 @@ def test_no_smash_all_diff_species(self):
         experiment.save()
 
-        mus_mus = Organism.get_object_for_name("MUS_MUSCULUS")
+        Organism.get_object_for_name("MUS_MUSCULUS")
```
I assume mus_mus was intended to be used here. Otherwise I can't see what this is helping to test.

arielsvn replied:
`Organism.get_object_for_name` also creates the organism in the db. Still it looks safe to delete the whole line since the organism `MUS_MUSCULUS` is not used after this.

arielsvn commented:
@cgreene thanks for the review, I addressed all your comments, and in some cases removed the whole line when the variable values didn't do anything.

@davidsmejia @kurtwheeler This should be ready for review now.

davidsmejia commented:
This looks really great. One question/concern though, would it be better to only remove unused imports and not remove unused variables in this PR. Looks like the tests will fail if an error is raised which is good but could be improved to make sure things aren't silently broken by checking that the functions/methods that return say a `job_context` is actually a `job_context` or if it returns a `processor_id` that the return value looks like a `processor_id`?

arielsvn commented:
> One question/concern though, would it be better to only remove unused imports and not remove unused variables in this PR.

I think there's value in removing unused variables as well, there're many that were removed that weren't doing anything.

> Looks like the tests will fail if an error is raised which is good but could be improved to make sure things aren't silently broken by checking that the functions/methods that return say a job_context is actually a job_context or if it returns a processor_id that the return value looks like a processor_id?

This would be also increasing the scope of this PR, but as you said is a good opportunity to also improve our tests. I was thinking just checking that the return values were not null. Is there anything else that you think we should assert @davidsmejia?

arielsvn commented:
Added additional assertions on https://github.com/AlexsLemonade/refinebio/pull/2052/commits/172abfc16f8842e3861976e0b69f16312569e3f7. @davidsmejia let me know if you think we should add any other.

davidsmejia commented on foreman/data_refinery_foreman/surveyor/test_harmony.py:
```diff
@@ -210,7 +210,8 @@ def test_sdrf_big(self):
             if not metadata:
                 continue
             # No assertions, just making sure we don't barf.
```
This comment can go!
